common/web/gesture-recognizer/src/engine/headless/gestures/matchers/gestureMatcher.ts

```diff
@@ -0,0 +1,293 @@
+import { ComplexGestureSource } from "../../complexGestureSource.js";
+import { SimpleGestureSource, SimpleGestureSourceSubview } from "../../simpleGestureSource.js";
+
+import { GestureModel, GestureResolution, GestureResolutionSpec, RejectionDefault, ResolutionItemSpec } from "../specs/gestureModel.js";
+
+import { ManagedPromise, TimeoutPromise } from "@keymanapp/web-utils";
+import { FulfillmentCause, PathMatcher } from "./pathMatcher.js";
+
+//
+// TODO:  Next up - let's unit-test this bad boy!
+//
+
+export interface MatchResult<Type> {
```

Comment:
I would like to encourage a doc comment on all exported things :grin: (I won't tag them all here though, so up to you)